### ISSUE TITLE
TypeVar fails to annotate as subscriptable

### ISSUE THREAD

**Issue by danielgordon10** [microsoft/pyright#1848]
Using 1.1.138

**To Reproduce**
```python
def foo(bar: Union[List, Tuple]) -> Union[List, Tuple]:
    val = bar[0]
    return bar
# No issue

T = TypeVar("T", List, Tuple)
def foo(bar: T) -> T:
    val = bar[0]
    return bar
# results in Object of type "List[Unknown] | Tuple[Unknown, ...]" is not subscriptablePylancereportGeneralTypeIssues
```

**Expected behavior**
No raised error like in Union case.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.140, which I just published. It will also be in the next release of pylance.
